src/lib/Libattr/attr_fn_str.c

```diff
@@ -308,3 +308,29 @@ decode_jobname(attribute *patr, char *name, char *rescn, char *val)
 	}
 	return (decode_str(patr, name, rescn, val));
 }
+
+/**
+ * @brief
+ * 	set_str_extnd - set attribute value based upon another
+ *
+ *	A=B --> A is replaced with B
+ *
+ * @param[in]   attr - pointer to new attribute to be set (A)
+ * @param[in]   new  - pointer to attribute (B)
+ * @param[in]   op   - operator
+ *
+ * @return      int
+ * @retval      0       if ok
+ * @retval     >0       if error
+ *
+ */
+
+int
+set_str_extnd(struct attribute *attr, struct attribute *new, enum batch_op op)
+{
```

Comment:
While what you say makes sense, this is a problem with our string resources as a whole.  I wouldn't single out partition and do something different.  We should either make an architectural decision to change set_str(), or leave it be.  I don't think partition should be different than other string resources.  This would cause a caveat to be placed in the docs to explain this.